internal/auth/authorizations.go

```diff
@@ -108,6 +117,7 @@ func QueryAuthd(authd, remoteIP, tlsEnabled, authSecret string) (*State, error)
 			switch p {
 			case "subscribe", "publish":
 			default:
+				// TODO: Can we add more granular permissions, for example delete, pause, etc?
 				return nil, fmt.Errorf("unknown permission %s", p)
 			}
```

Comment:
If an auth server returns additional permissions it would cause `unknown permission` to occur on previous versions of `nsqd`. Do you think this should be relaxed, for example log the unknown permission and move on?

Another thing I thought about was a sort of "feature negotiation" with the auth server; either passing the `nsqd` or `nsqlookupd` version or passing the set of permissions it's able to handle. We could delegate the responsibility of trimming the list and/or erroring to the auth server. I'm not sold this is worthwhile, but without some kind of change (relaxing or negotiation) the auth server and `nsqd` versions will need to be kept in sync - unless a different `secret` is used by the client.